sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BlobQueryReader.java

```diff
@@ -74,6 +77,32 @@ public Flux<ByteBuffer> read() {
             .concatMap(this::parseRecord);
     }
 
+    /**
+     * Avro parses a query reactive stream.
+     *
+     * The Avro stream is formatted as the Avro Header (that specifies the schema) and the Avro Body (that contains
+     * a series of blocks of data). The Query Avro schema indicates that the objects being emitted from the parser can
+     * either be a result data record, an end record, a progress record or an error record.
+     *
+     * @return The parsed query reactive stream.
+     */
+    public InputStream readInputStream(InputStream inputStream) throws IOException {
+        // Convert InputStream to Flux<ByteBuffer>
+        Flux<ByteBuffer> avroFlux = toFluxByteBuffer(inputStream);
+
+        // Use existing read method to process the data
+        Flux<ByteBuffer> processedData = new AvroReaderFactory().getAvroReader(avroFlux).read()
+            .map(AvroObject::getObject)
+            .concatMap(this::parseRecord);
+
+        return new FluxInputStream(processedData);
+    }
+
+    private Flux<ByteBuffer> toFluxByteBuffer(InputStream inputStream) {
```

Comment:
Is this method needed? Can't the method above just call `FluxUtil.toFluxByteBuffer` directly?